### PR TITLE
TNT-38925 - send_notifications implementation

### PR DESCRIPTION
**Pull request by dcottingham**
1. TargetClient.send_notifications(options) will send click or display notifications to Delivery API for metrics calculations.
2. Fix CI test run to run against all configured versions of python
3. Make all code backward compatible with python 2.7

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,28 +16,34 @@ new_env: clean_env pre_env env
...
 	rm -rf build dist target_python_sdk.egg-info
 
-# Install dependencies - pip update sometimes required for python 2.7
+# Install dependencies - older versions of python need to bootstrap pip for CI workflow
 install: env
-	curl https://bootstrap.pypa.io/get-pip.py | python
-	pip install --upgrade pip "setuptools>=38.3.0,<=44.1.x" wheel pep517 pylint twine bump2version coverage coveralls
-	pip install -r requirements.txt
-	pip install -r test-requirements.txt
+ifneq (,$(filter $(PYTHON_VERSION),2.7 3.4 3.5))
```
Comment (5amfung) [inline Makefile:21]: I'm surprised the github workflow image doesn't already include a proper pip.

File: target_python_sdk/tests/helpers.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2020 Adobe. All rights reserved.
```
Comment (5amfung) [inline target_python_sdk/tests/helpers.py:1]: 2021

File: target_python_sdk/tests/responses/notifications.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  "edgeHost": "mboxedge28.tt.omtrdc.net",
+  "notifications": [
+      {
+          "id": "SummerOfferNotification"
+      },
+      {
+          "id": "SummerShoesOfferNotification"
+      }
+  ]
+}
```
Comment (5amfung) [inline target_python_sdk/tests/responses/notifications.json:17]: EOF newline

**Review by 5amfung — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (dcottingham) [reply Makefile:21]: It includes the latest version, which doesn't work for older python builds

**Review by jasonwaters — APPROVED** [review]
Looks good @dcottingham !  Apologies for the delay in review.

**Pull request merged**